File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False

        char_map = {}

        for c in range(ord("a"), ord("z") + 1):
            char_map[chr(c)] = 0

        for i in range(len(s)):
            char_s = s[i]
            char_t = t[i]

            char_map[char_s] = char_map[char_s] + 1
            char_map[char_t] = char_map[char_t] - 1

        for count in char_map.values():
            if count != 0:
                return False

        return True

File: 0242-valid-anagram/0242-valid-anagram.java
 

class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) {
            return false;
        }

        HashMap<Character, Integer> map = new HashMap<>();

        for (char c = 'a'; c <= 'z'; c++) {
            map.put(c, 0);
        }

        for (int i = 0; i < s.length(); i++) {
            char charS = s.charAt(i);
            char charT = t.charAt(i);

            map.put(charS, map.get(charS) + 1);
            map.put(charT, map.get(charT) - 1);
        }

        for (int count : map.values()) {
            if (count != 0) {
                return false;
            }
        }

        return true;
    }
}